hoomd/GPUArray.h

```diff
@@ -934,7 +934,6 @@ template<class T> void GPUArray<T>::allocate()
     h_data = std::unique_ptr<T, hoomd::detail::host_deleter<T> >(reinterpret_cast<T *>(host_ptr), host_deleter);
 
 #if defined (ENABLE_HIP)
-    assert(!d_data);
```

Comment:
Bottom line is that the assert is old and should have been removed a while ago, but was never done since everyone uses release builds of hoomd. This was caught during unit testing when I was using a debug build, which actually checks the assert statements.